### ISSUE TITLE
sanitize_param() not correctly resolving when using nested columns

### ISSUE THREAD

**Issue by Vidski** [supabase-community/supabase-py#203]
**Describe the bug**
When filtering for a nested column, sanitize_param is adding unneccessary " "

**To Reproduce**
`supabase.table('character_inventory').select('character(*), slot, item(*, template(*))').eq('**character.user_id**', user_id).execute()`

resolves to:
`/rest/v1/character_inventory?select=character(*),+slot,+item(*,+template(*))&**"character.user_id"**=eq.2731f9cc-17f4-42a2-839a-d25dfe8d90a8` returns HTTP 400

**Expected behavior**
Should resolve to:
`/rest/v1/character_inventory?select=character(*),+slot,+item(*,+template(*))&**character.user_id**=eq.2731f9cc-17f4-42a2-839a-d25dfe8d90a8` returns HTTP 200

**Screenshots**
![image](https://user-images.githubusercontent.com/36316706/166309069-a07390a7-c1c8-4aa1-9d54-c6fbcf5234d5.png)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Version: latest

**Comment by Vidski** [comment]
![image](https://user-images.githubusercontent.com/36316706/166310268-e5cc72ef-a7f9-4f2c-85fb-c9e29fd92817.png)

**Comment by Vidski** [comment]
Is there a reason to sanitize the column at all? In the postgrest-js client all we do is remove whitespaces.

**Comment by anand2312** [comment]
> Is there a reason to sanitize the column at all? In the postgrest-js client all we do is remove whitespaces.

Column names can contain postgrest special characters which need to be escaped, The js client errors out because of this; see supabase/postgrest-js#262

Are you sure you are on the latest version though? The code in that screenshot doesn't look like it is from the latest version

**Comment by Vidski** [comment]
I'm fine with escaping ',:()', but escaping '.' results in a wrong behaviour which makes it impossible to filter on foreign columns.

eq('table_a.table_b.id', my_id) will currently resolve to "table_a.table_b.id", but should resolve to "table_a"."table_b"."id". 
Maybe we should first split the column by '.', escape each item, and join using '.'.

This is using supabase-py 0.5.5 and postgrest-py 0.10.2

**Comment by anand2312** [comment]
Yeah I get what you mean 👍🏼 this'll be fixed in the next release. Thanks for the report!

**Comment by J0** [comment]
Moving to the postgrest-py repo
